debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/AbstractReader.java

```diff
@@ -69,7 +70,15 @@ public void uponCompletion(Runnable handler) {
     }
 
     @Override
+    public void initialize() {
+        if (this.initialized.compareAndSet(false, true)) {
+            doInitialize();
+        }
+    }
+
+    @Override
     public void start() {
+        initialize();
```

Comment:
It's surprising that it is called here by `start()`. I'd rather have expected that the task or connector calls it before calling `start()` (as you actually do it below)?

Reply:
It needs to be called, and I debated about not calling it here. But all current implementation handle calling more than once, so I thought this was safer. But I don’t have a preference.

Reply:
Do you mean to say "It needs *not* to be called"? If so, I'd prefer to not do it.

Reply:
Sorry, meant to say that `initialize()` needs to be called at some point, either explicitly by the user or, if we wanted, implicitly in the `start()` method.